### ISSUE TITLE
`--stdin` option should also allow passing filename

### ISSUE THREAD

**Issue by derekprior** [bbatsov/rubocop#3972]
I'm using [ALE](https://github.com/w0rp/ale/) to lint files asynchronously from Vim. This uses the `--stdin` flag to lint the file as it exists in the buffer, which may be different from the file on disk. When using rubocop this approach breaks down when editing a file that matches a configured excluded path or when editing a file in a subfolder that has its own `.rubocop.yml`. 

If we could pass the filename of the code being linted on `STDIN` this could be alleviated. ESLint, for example, offers `--stdin-filename` for this purpose.

-----

## Expected behavior

`exclude`s and other filename based behavior should be supportable when using `--stdin`

## Actual behavior

`--stdin` has no way to specify the filename, so no exclusions are applied.

## Steps to reproduce the problem

Use this rubocop config:

```yml
AllCops:
  Exclude:
    - 'db/schema.rb'
```

Add a `db/schema.rb` file with linting errors.

Run `rubocop --stdin` to lint the contents of `db/schema.rb`

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.47.1
```

**Comment by s-ol** [comment]
I think this is already working. My test case:

```
 $ cat .rubocop.yml                                                                       
AllCops:
  Exclude:
    - 'test.rb'
$ cat test.rb                                                                            
3 = 3
$ cat test.rb | rubocop --force-exclusion --stdin test.rb                                
Inspecting 0 files


0 files inspected, no offenses detected
$ rubocop --force-exclusion test.rb                                                      
Inspecting 0 files


0 files inspected, no offenses detected
$ rubocop test.rb                                                                        
Inspecting 1 file
E

Offenses:

test.rb:1:3: E: unexpected token tEQL
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
3 = 3
  ^

1 file inspected, 1 offense detected
```

**Comment by jonas054** [comment]
That's right. The `--stdin` option takes a file name argument, but this is not documented in the [manual](https://github.com/bbatsov/rubocop/blob/v0.47.1/manual/basic_usage.md) or in the output from `rubocop --help`.

**Comment by derekprior** [comment]
Thanks for the clarification. I verified this works. I took a quick look at how to add this information to `--help` output but wasn't sure.

**Comment by jonas054** [comment]
Let's keep the issue open until we have updated the documentation.

**Comment by yash-nisar** [comment]
This can be closed since we have updated the documentation. CC @jonas054
